src/main/java/com/google/sps/servlets/UploadMemes.java

```diff
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import com.google.api.gax.paging.Page;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import java.util.*;
+
+@WebServlet("/memes-upload")
+@MultipartConfig
+public class UploadMemes extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Since you'll also use this servlet to "get" memes and not just upload, I suggest renaming the file/class/URL to just be:
* file = MemesServlet.java
* class = MemesServlet
* url = "/memes"